File: nomad-samples/js/asset-upload/assets/single-thread-upload.js
import uploadPart from "./upload-part.js";
import uploadPartComplete from "./upload-part-complete.js";

export default async function singleThreadUpload(AUTH_TOKEN, FILE, UPLOAD_RESPONSE) {
    // Loop all parts
    for (let index = 0; index < UPLOAD_RESPONSE.parts.length; index++) {
        console.log(`Uploading part ${index + 1} of ${UPLOAD_RESPONSE.parts.length}...`);
        // Set the current part
        const part = UPLOAD_RESPONSE.parts[index];

        // Upload this part
        const etag = await uploadPart(FILE, part.url, part);

        // Complete this part upload
        await uploadPartComplete(AUTH_TOKEN, part.id, etag);

        console.log(`Uploaded part ${index + 1} of ${UPLOAD_RESPONSE.parts.length} successfully`);
    }
}

File: nomad-samples/common/py/asset_upload/assets/single_thread_upload.py
from assets.upload_asset_part_complete import *
from assets.upload_asset_part import *

async def single_thread_upload(AUTH_TOKEN, FILE, RESPONSE):
    TOTAL_PARTS = len(RESPONSE["parts"])

    for IDX, PART in enumerate(RESPONSE["parts"]):
        print(f"Uploading part {IDX + 1} of {TOTAL_PARTS}...")

        ETAG = await upload_asset_part(FILE, PART["url"], PART)

        await upload_asset_part_complete(AUTH_TOKEN, PART["id"], ETAG)

        print(f"Uploaded part {IDX + 1} of {TOTAL_PARTS} successfully")